mocha_test/slice.js

```diff
@@ -0,0 +1,32 @@
+var slice = require('../lib/internal/slice').default;
```

Comment:
Yes(ish), as `concat` now uses `internal/slice`. I wanted to make sure the behaviour `concat` relies on isn't broken by accident. Plus, `slice` is used in a few of our functions, so it's good to have some tests for it similar to the DLL [ones](https://github.com/caolan/async/blob/master/mocha_test/linked_list.js).